WHID_Elite_HybridCC_v.1.0/WHID_Elite_HybridCC_v.1.0.ino

```diff
@@ -0,0 +1,463 @@
+// ###################################################################################################
+// #                                      WHID Elite Hybrid 1.0.0                                  ###
+// #                              Copyright 2018-2019 - @LucaBongiorni                             ###
+// #                              Fork 2020 HybridC2 - @JoelSernaMoreno                            ###
+// ###################################################################################################
+// ##################################### Commands Available ##########################################
+// ###################################################################################################
+// ### batt:                         ### Example: ###   batt:                                        #
+// ### enumos:                       ### Example: ###   enumos:                                      #
+// ### airgapwin:<COMMAND-TO-EXFIL>  ### Example: ###   airgapwin:whoami                             #
+// ### airgapnix:<COMMAND-TO-EXFIL>  ### Example: ###   airgapnix:whoami                             #
+// ### airgaposx:<COMMAND-TO-EXFIL>  ### Example: ###   airgaposx:whoami                             #
+// ### win:<COMMAND>                 ### Example: ###   win:iexplore -k http://fakeupdate.net/wnc/   #
+// ### nix:<COMMAND>                 ### Example: ###   nix:gnome-calculator                         #
+// ### osx:<COMMAND>                 ### Example: ###   osx:open -n -a calculator                    #
+// ### spy:<PHONE-NUMBER>            ### Example: ###   spy:0039123123123                            #
+// ### getlocation:                  ### Example: ###   getlocation:                                 #
+// ### hybridC2:                     ### Example: ###   hybridC2:                                    #
+// ### hybridC2off:                  ### Example: ###   hybridC2off:                                 #
+// ###################################################################################################
+
+#include "Adafruit_FONA.h"
+#include "FingerprintUSBHost.h"
+#include <Keyboard.h>
+
+int dontSMS=0; // Set to 1 to save SMS
+int dontGPRS=0; // Set to 1 to use GPRS
+char gprsAirgap[50];
```

Comment:
I guess also here should be encreased to 5000 for testing.

Reply:
I answer myself... declaring it to 5000 will be too much. Not Enough memory left while compiling the sketch.

Reply:
Leaving this here as reminder: https://learn.adafruit.com/memories-of-an-arduino/optimizing-sram

Reply:
_**Example:**_
**With char gprsAirgap[1] we get....**
Sketch uses 20606 bytes (71%) of program storage space. Maximum is 28672 bytes.
Global variables use _**1617 bytes (63%) of dynamic memory, leaving 943 bytes for local variables.**_ Maximum is 2560 bytes.

**With char gprsAirgap[500] we get...**
Sketch uses 20654 bytes (72%) of program storage space. Maximum is 28672 bytes.
Global variables use _**2116 bytes (82%) of dynamic memory, leaving 444 bytes for local variables. Maximum is 2560 bytes.**_
Low memory available, stability problems may occur.

Reply:
**I am trying to think of a solution...** 
Maybe a payload that executes the command to exfil (e.g. "ifconfig") and then it splits the output and send over serial in multiple chunks with delay between each other delivery?

Reply:
Or.... we can create a payload that splits the output in chunks of 488 chars and append at the end a special terminator that is used to tell whidleite to send the POST and restart listening the serial... by using something like... `String BypassAirGap = Serial.readStringUntil(':TERMINATOR:');`

Did you get my point?